server/src/main/java/org/elasticsearch/common/util/LongHash.java

```diff
@@ -41,7 +41,13 @@ public LongHash(long capacity, BigArrays bigArrays) {
     //Constructor with configurable capacity and load factor.
     public LongHash(long capacity, float maxLoadFactor, BigArrays bigArrays) {
         super(capacity, maxLoadFactor, bigArrays);
-        keys = bigArrays.newLongArray(capacity, false);
+        try {
+            keys = bigArrays.newLongArray(capacity, false);
+        } finally {
```

Comment:
Maybe it is worthy a comment explaining that the parent class allocates a BigArray and therefore if this allocation fails, we need to make sure we release objects of the parent class?

Reply:
:+1: